.teamcity/src/main/kotlin/configurations/CompileAll.kt

```diff
@@ -1,30 +1,59 @@
 package configurations
 
+import configurations.CompileAll.BuildCacheType.PRODUCTION_BUILD_CACHE
 import model.CIBuildModel
 import model.Stage
 
-class CompileAll(model: CIBuildModel, stage: Stage) : BaseGradleBuildType(stage = stage, init = {
-    id(buildTypeId(model))
-    name = "Compile All"
-    description = "Compiles all production/test source code and warms up the build cache"
+class CompileAll(model: CIBuildModel, stage: Stage, buildCacheType: BuildCacheType = PRODUCTION_BUILD_CACHE) :
+    BaseGradleBuildType(stage = stage, failStage = buildCacheType.failStage, init = {
+        id(buildTypeId(model, buildCacheType))
+        name = "Compile All${buildCacheType.nameSuffix}"
+        description = "Compiles all production/test source code and warms up the build cache"
 
-    features {
-        publishBuildStatusToGithub(model)
-    }
+        features {
+            publishBuildStatusToGithub(model)
```

Comment:
I don't think the build cache NG compile all should publish the status to github. It may show a failed check for folks and most of the devs shouldn't care about it.